pkg/util/etcdutil/migrate.go

```diff
@@ -62,6 +64,11 @@ func Migrate(kubeConfig clientcmd.ClientConfig, etcdServiceIP string) error {
 	}
 	glog.Info("etcd cluster for migration is now running")
 
+	etcdServiceIP, err := getServiceIP(kubecli, api.NamespaceSystem, asset.EtcdServiceName)
```

Comment:
can we do this at line51? it is cleaner to get all prerequisites at one place before we start the migration.